File: OpenCVによる画像処理入門/第3版/第1刷/sampleprogram/5-1-2.cpp
#define _USE_MATH_DEFINES
#include <iostream>
#include <cmath>
#include <opencv2/opencv.hpp>
using namespace std;
using namespace cv;
string win_src = "src";
string win_dst = "dst";

int main()
{
  string file_src = "src.png"; // 入力画像のファイル名
  string file_dst = "dst.png"; // 出力画像のファイル名
  Mat img_src = imread(file_src, 1); // 入力画像（カラー）の読み込み

  Mat img_dst;
  if(!img_src.data) {
    cout << "error" << endl;
    return -1;
  }

  // ウインドウ生成
  namedWindow(win_src, WINDOW_AUTOSIZE);
  namedWindow(win_dst, WINDOW_AUTOSIZE);

  // 複数色チャンネルの分割
  vector<Mat> img_bgr(3);
  split(img_src, img_bgr);

  // 青→赤，緑→青，赤→緑に変更
  merge(vector<Mat>{img_bgr[1], img_bgr[2], img_bgr[0]}, img_dst);

  imshow(win_src, img_src); // 入力画像を表示
  imshow(win_dst, img_dst); // 出力画像を表示
  imwrite(file_dst, img_dst); // 処理結果の保存

  waitKey(0); // キー入力待ち
  return 0;
}

File: OpenCVによる画像処理入門/第3版/第1刷/sampleprogram/5-1-2.py
import cv2
import math
import numpy as np

file_src = 'src.png'
file_dst = 'dst.png'

img_src = cv2.imread(file_src, 1)

cv2.namedWindow('src')
cv2.namedWindow('dst')

# 複数色チャンネルの分割
img_bgr = cv2.split(img_src)
# 青→赤，緑→青，赤→緑に変更
img_dst = cv2.merge((img_bgr[1], img_bgr[2], img_bgr[0]))

cv2.imshow('src', img_src)  # 入力画像を表示
cv2.imshow('dst', img_dst)  # 出力画像を表示
cv2.imwrite(file_dst, img_dst)  # 処理結果の保存

cv2.waitKey(0)  # キー入力待ち
cv2.destroyAllWindows()
